Make udev link re-renaming conditional

keszybz opened the pull request.

evverx commented:
This pull request **introduces 1 alert** when merging 75b13196120116783394db2b3ae2ccc521401779 into 35f2c0ba6afd70c4a1f74865d9231b85f4a01380 - [view on LGTM.com](https://lgtm.com/projects/g/systemd/systemd/rev/pr-92b3a6e3582b5d06e522acf49a08c9f3f9ff52e6)

**new alerts:**

* 1 for Missing header guard

---

*Comment posted by [LGTM.com](https://lgtm.com)*

keszybz commented:
Updated to add missing pragma.

poettering commented on src/udev/net/link-config.c:
```diff
@@ -352,12 +333,13 @@ static int get_mac(sd_device *device, bool want_random,
         const char *new_name = NULL;
         const char *old_name;
         unsigned speed;
+        unsigned name_type = NET_NAME_UNKNOWN;
```
merge with previous line?

poettering commented on src/udev/net/link-config.c:
```diff
@@ -410,42 +392,56 @@ int link_config_apply(link_config_ctx *ctx, link_config *config,
-                                        break;
-                                default:
-                                        break;
+        (void) link_name_type(device, &name_type);
```
i think adding debug logging here if this fails would make a ton of sense

keszybz replied:
It has. `link_name_type` logs on it's own. (That makes more sense because there's a two failure points.)

poettering replied:
a true. ignore me.

poettering commented on src/udev/net/link-config.c:
```diff
@@ -410,42 +392,56 @@ int link_config_apply(link_config_ctx *ctx, link_config *config,
-                if (!new_name)
-                        new_name = config->name;
+        if (new_name)
+                log_device_info(device, "Policy *%s* yields \"%s\".", name_policy_to_string(policy), new_name);
```
maybe LOG_DEBUG?

poettering commented on src/udev/net/naming-scheme.c:
```diff
@@ -0,0 +1,65 @@
+};
+
+
+static const NamingScheme* naming_scheme_from_name(const char *name) {
```
spurious double newline

poettering commented:
i think a small NEWS additoinal about this would be good.

keszybz commented:
All updated, PTAL.

yuwata commented:
Still two `log_device_info()`. That is not strong opinion, but why?

yuwata commented:
Please update the following:
https://github.com/systemd/systemd/blob/850115b3a1119c87f14ecb7634615632ef0b1933/man/systemd.link.xml#L261-L262

keszybz commented:
Updated. Sorry about the missed log levels: I updated on of the patches and forgot that the other one needs updating too. I changed the two remaining log_info's to log_debug as suggested by @yuwata. There doesn't seem to be a good reason to make those two different than the surrounding ones.

I also added a formatting patch on top make the man page nicer.

martinpitt commented:
This brought back the naming regression: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=917128#72 . This is using systemd .netlink files with `Name=` assignments.

yuwata commented:
@martinpitt Does 03dd7b7ddec1b0e06f254972a2e05f516a05edaf (#11729) fix the issue? BTW, please file the issue as a new issue.
